### PR TITLE
Made world data paths adjustable, and added API to temporarily disable saving chunks to disk.

### PR DESCRIPTION
**Pull request by lkolbly**
This PR makes the data path that world data is saved in adjustable at world creation time (via constructor parameter and settings.ini parameter), which will be useful for restructuring the world files (as in #3426). The Anvil format, player stats, and maps are all modified to accommodate this change.

It adds the optional [WorldPaths] section to the settings.ini file, which specifies the directories that various worlds are stored in, for example:
```
[WorldPaths]
world=Worlds/overworld
world_nether=Worlds/nether
world_the_end=Worlds/end
```

This PR also adds the SetSavingEnabled method to cWorld and cRoot, which permits disabling saving chunk files to disk (note that it doesn't prevent writing the ini file to disk). This allows implementing the /save-off and /save-on commands. It will also allow easily resetting maps, I think will be useful for minigames (resetting the map then just becomes an issue of reloading the world).

m_SavingEnabled is atomic, so that cRoot can access it from its thread safely for console commands.

Tweaked the HOOK_CHUNK_UNLOADING documentation to say that the ability for the plugin to actually prevent the chunk from unloading is a feature not a bug.

### CODE REVIEW

**Comment by madmaxoft** [comment]
I'm not too happy about having the world config split into two separate sections. But I can't think of a better way.

**Comment by lkolbly** [comment]
Ditto. On further thought, maybe another idea would be:
```
[Server]
...
DefaultWorld=world

[Worlds]
world=Worlds/world
world_nether=Worlds/world_nether
```

Where [Worlds] just specifies the paths, but then the [Server] section gets muddied with the DefaultWorld tag.

**Comment by lkolbly** [comment]
The [WorldPaths] is optional, so it should only appear in the rare instances where the paths are not the default. (the default for <name> would be Worlds/<name> for the architecture change)

**Comment by madmaxoft** [comment]
Migration to that format would be non-trivial.

If only we already had Lua-based config files, we could simply do:
```lua
Worlds =
{
  {
    Name = "world",
    Path = "Worlds/world",
    SomeOtherParam = 3.1415,
  },
  -- ...
}
```

**Comment by lkolbly** [comment]
If I make the config files lua-based (in a future PR), can I make the crafting/brewing/etc files lua-based too?

**Comment by madmaxoft** [comment]
Yes, but please discuss this first in the forum, we have had some discussions there already and there are some ideas that you might want to consider before diving in :)

File: src/Root.h [hunk 1]
```diff
@@ -137,6 +137,9 @@ class cRoot
 	/** Saves all chunks in all worlds */
 	void SaveAllChunks(void);  // tolua_export
 
+	/** Sets whether saving chunks is enabled */
```
Comment (madmaxoft) [inline src/Root.h]: `for all worlds`

File: src/World.h [hunk 2]
```diff
@@ -111,6 +112,12 @@ class cWorld :
 
 	// tolua_begin
 
+	/** Get whether saving chunks is enabled */
+	bool GetSavingEnabled(void) const { return m_SavingEnabled; }
```
Comment (madmaxoft) [inline src/World.h]: Rename to `IsSavingEnabled` and `m_IsSavingEnabled`

File: src/World.h [hunk 3]
```diff
@@ -871,6 +881,9 @@ class cWorld :
 
 	AString m_WorldName;
 
+	/** The path to the root directory for the world files */
```
Comment (madmaxoft) [inline src/World.h]: `Doesn't contain the trailing path separator.`

File: Server/Plugins/APIDump/Hooks/OnChunkUnloading.lua [hunk 4]
```diff
@@ -8,8 +8,7 @@ return
 			Cuberite calls this function when a chunk is about to be unloaded from the memory. A plugin may
 			force Cuberite to keep the chunk in memory by returning true.</p>
 			<p>
-			FIXME: The return value should be used only for event propagation stopping, not for the actual
-			decision whether to unload.
+			CAUTION: Preventing the server from unloading chunks can cause the server to use too much RAM, which will adversly affect both performance and stability (i.e. your computer will get slow and crash). Return true sparingly.
```
Comment (lkolbly) [inline Server/Plugins/APIDump/Hooks/OnChunkUnloading.lua]: Today I learned.

**Comment by lkolbly** [comment]
Alright, made some fixes and pulled the latest master.

I'll start working on Lua config files once all the file architecture dust settles. :-)

**Comment by lkolbly** [comment]
Alright, _now_ it passes CI.

**Comment by bearbin** [comment]
Resolved the conflict, just waiting for CI to pass then merge.

**Pull request merged**